clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -492,14 +999,21 @@ void SemaHLSL::handleResourceBindingAttr(Decl *D, const ParsedAttr &AL) {
   // Validate.
   if (!Slot.empty()) {
     switch (Slot[0]) {
```

Comment:
Can `getRegisterTypeIndex` be reused here?  Seems like there's more than one table of valid register types now.

Reply:
I've cleaned up a bit, and realized I no longer needed to pass `Slot`, when I could just pass registerType instead. 
`getRegisterTypeIndex` is still needed for `ValidateMultipleRegisterAnnotations`, but all other calls to `getRegisterTypeIndex` have been removed, and I baked in the logic `getRegisterTypeIndex` has into the initial case statement in `handleResourceBindingAttr`.

Reply:
This still seems to be essentially a duplicate of `getRegisterTypeIndex`.  You may need to modify `getRegisterTypeIndex` a bit to make it work here, but really that would be much better than duplicating the switch statement like this.